app/models/vote_account_history.rb

```diff
@@ -34,4 +35,52 @@
 #
 class VoteAccountHistory < ApplicationRecord
   belongs_to :vote_account
+
+  after_create :set_skipped_vote_percent_moving_average
+
+  def previous_24_hours
+    self.class.where(vote_account: vote_account, network: network, created_at: 24.hours.ago..created_at)
+  end
+
+  def skipped_vote_percent
+    ((slot_index_current.to_i - credits_current.to_i)/slot_index_current.to_f).round(2)
+  end
+
+  def set_skipped_vote_percent_moving_average
+    previous_24_hours_set = self.previous_24_hours.to_a
+    if previous_24_hours_set.count > 0
+      self.skipped_vote_percent_moving_average = previous_24_hours_set.map(&:skipped_vote_percent).sum/previous_24_hours_set.count
+    end
+    save
+  end
+
+  def self.average_skipped_vote_percent_for(network:, batch_uuid:)
+    vah_skipped = where(
+      network: network,
+      batch_uuid: batch_uuid
+    ).map(&:skipped_vote_percent)
+    vah_skipped.sum / vah_skipped.count
+  end
+
+  def self.median_skipped_vote_percent_for(network:, batch_uuid:)
+    vah_skipped = where(
+      network: network,
+      batch_uuid: batch_uuid
+    ).map(&:skipped_vote_percent)
+    vah_skipped.sort[vah_skipped.length/2]
+  end
+
+  def self.average_skipped_vote_percent_moving_average_for(network:, batch_uuid:)
+    where(
+      network: network,
+      batch_uuid: batch_uuid
+    ).average(:skipped_vote_percent_moving_average)
+  end
+
+  def self.median_skipped_vote_percent_moving_average_for(network:, batch_uuid:)
```

Comment:
This isn't used/called from anywhere. What's it intended for? Please leave a comment for what it's going to be used for in the future if it really is needed.